src/rendering/RenderUtil.cc

```diff
@@ -1070,13 +1138,14 @@ void RenderUtilPrivate::UpdateRenderingEntities(
         return true;
       });
 
-  // lights
+  // update lights
   _ecm.Each<components::Light, components::Pose>(
       [&](const Entity &_entity,
-        const components::Light *,
+        const components::Light *_light,
         const components::Pose *_pose)->bool
       {
         this->entityPoses[_entity] = _pose->Data();
+        this->entityLights[_entity] = _light->Data();
```

Comment:
I'm thinking of making this more efficient. Currently we are populating `entityLights` for all lights in the scene and going through the update process every iteration. Even though we check to see if any light properties have changed, there is still a bit of computation overhead every iteration. In an upcoming environment we are creating, we plan to put in a few hundred lights so this may add up. 

How about we handle light requests similar to pose commands by adding a new `components::LightCmd`? When the light service request is received, we create a `LightCmd` component and attach it to the light entity in UserCommands, e.g. see `WorldPoseCmd` creation [here](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/src/systems/user_commands/UserCommands.cc#L671). In RenderUtil, we will instead be looping through entities with `components::LightCmd`:

```
_ecm.Each<components::LightCmd>(
      [&](const Entity &_entity,
        const components::LightCmd *_light) -> bool
   this->entityLights[_entity] = _light->Data();
   ...
```` 

We then remove the `components::LightCmd` from the entity afterwards to indicate that we've processed the cmd, e.g. see removing `WorldPoseCmd` [here](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/src/systems/physics/Physics.cc#L1659)

Reply:
@iche033 I added most of your feedback, but I'm not sure where I should remove the LightCmd. The worldPoseCmd is remove in Physics, but  where is the right place to remove these cmds? renderutils.cc?

Reply:
@iche033 I removed the LightCmd, I had to convert the const-EntityComponentManager vriable in to a non-const EntityComponentManager variable to be able to remove the LightCmd. is this correct ?

https://github.com/ignitionrobotics/ign-gazebo/blob/19b3c9f06d5c097dc923324501a2a05bb889a2dd/src/rendering/RenderUtil.cc#L1162-L1169

Reply:
hmm I see. `UpdateFromECM` is called in the [PostUpdate](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/src/systems/sensors/Sensors.cc#L437) function in the `sensors` system on the server side in the sensors system, which gets a const ref `EntityComponentManager`. The physics system is able to remove the component as it's done in [Update](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/src/systems/physics/Physics.cc#L1659) function which has a mutable reference to ECM. We may have to create a new `UpdateECM` function in RenderUtil and call it from a new `Update` function (which also needs to be added)  in `sensors` system. 

On the gui side, the [UpdateFromECM ](https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo4/src/gui/plugins/scene3d/Scene3D.cc#L2411) is called in a function that has a mutable reference to ECM. So we should be able to just do `UpdateECM()` and followed by `UpdateFromECM`

Reply:
@iche033 done https://github.com/ignitionrobotics/ign-gazebo/pull/482/commits/0e660f93578f097fd8e72337febb3009eeebacad